src/khepri_machine.erl

```diff
@@ -1400,8 +1554,36 @@ overview(State) ->
       triggers => Triggers,
       keep_while_conds => KeepWhileConds}.
 
+-spec version() -> MacVer when
+      MacVer :: 1.
+%% @doc Returns the state machine version.
+
 version() ->
-    0.
+    1.
+
+-spec which_module(MacVer) -> Module when
+      MacVer :: 1 | 0,
+      Module :: ?MODULE.
+%% @doc Returns the state machine module corresponding to the given version.
+
+which_module(1) -> ?MODULE;
+which_module(0) -> ?MODULE.
+
+-spec effective_version(StoreId) -> Ret when
+      StoreId :: khepri:store_id(),
+      Ret :: khepri:ok(EffectiveMacVer) | khepri:error(),
+      EffectiveMacVer :: ra_machine:version().
+%% @doc Returns the effective state machine version of the local Ra server.
+
+effective_version(StoreId) ->
+    ThisNode = node(),
+    RaServer = khepri_cluster:node_to_member(StoreId, ThisNode),
+    case ra:member_overview(RaServer) of
+        {ok, #{effective_machine_version := EffectiveMacVer}, _LeaderId} ->
+            {ok, EffectiveMacVer};
+        {error, _} = Error ->
+            Error
+    end.
```

Comment:
Once we update Ra to a version that includes https://github.com/rabbitmq/ra/pull/426 I believe this should become:

```suggestion
    case ra_counters:counters(RaServer, [effective_machine_version]) of
        #{effective_machine_version := EffectiveMacVer} ->
            {ok, EffectiveMacVer};
        _ ->
            {error, ?khepri_error(effective_machine_version_not_defined,
                                  #{ra_server => RaServer})}
    end.
```

Reply:
What about calling `ra:member_overview/1` as a fallback if the counter doesn't provide the version?

Reply:
The counter is set up when the server process initializes so there should be no case where a `ra:member_overview/1` would return successfully while a counters lookup fails

Reply:
One more thing: I believe we don't need to wait for a new release of Ra that contains the new counter, the code should be compatible with older versions of Ra, shouldn't it?

Reply:
Ah yep. If we fall back to using the `ra:member_overview/1` then we wouldn't have to depend on the most recent changes in Ra :+1:

Reply:
It works locally when used with an old version of Ra. I committed your suggestion with the fallback to `ra:member_overview/1`.